### ISSUE TITLE
Next major version

### ISSUE THREAD

**Issue by puranjayjain** [puranjayjain/md-date-time-picker#215]
# Next major step
There are loads of problems with the current version
* Bigger download footprint (where not all of the features are not used by all users)
* Loads of bugs and confusion in the current implementation
* Introduction of material web components (far more promising than mdl)

# Solution
I personally prefer to write a new version using the material components as the baseline e.g using the dialog etc. modules. This would mean we can reduce the size footprint and make our component modular and in line with the modern bundlers to make it more efficient to use in any project.

What are your thoughts/suggestions/anything else @redbmk @westy92 @digitalkaoz @ggomagundan @codingfriendsnippets

**Comment by redbmk** [comment]
If this is going to have a dependency on material web components it might be nice to try making it an official component.

**Comment by puranjayjain** [comment]
@redbmk I also want that but getting a component through would be far more time consuming and even if we push for it from now we may or may not see it live in 6 or more months. I rather see this live than get work hours wasted

**Comment by westy92** [comment]
I agree with @redbmk that we should build this as a Material Design Web component.  Here's the relevant issues:

https://github.com/material-components/material-components-web/issues/55
https://github.com/material-components/material-components-web/issues/54

**Comment by redbmk** [comment]
Maybe the first push could be to get it live using material web components as a dependency, then afterwards submit it as a PR to the official team

**Comment by puranjayjain** [comment]
this would be a better approach
